css-shapes-2/Overview.bs

```diff
@@ -162,17 +172,32 @@ The ''shape()'' Function</h4>
 	Each command's starting point is the previous command's ending point.
 
 	<pre class=prod>
-		<<shape-command>> = <<move-command>> | <<line-command>> | <<hv-line-command>> |
-		                  <<curve-command>> | <<smooth-command>> | <<arc-command>> | close
+		<<shape-command>> = <<move-command>> | <<line-command>> | close
+		                  <<horizontal-line-command>> | <<vertical-line-command>> | 
+		                  <<curve-command>> | <<smooth-command>> | <<arc-command>>
+
+		<<move-command>> = move <<command-end-point>>
+		<<line-command>> = line <<command-end-point>>
+		<<horizontal-line-command>> = hline
+						[ to [ <<length-percentage>> | left | center | right | x-start | x-end ]
+						| by <<length-percentage>> ]
+		<<vertical-line-command>> = vline
+						[ to [ <<length-percentage>> | top | center | bottom | y-start | y-end ]
+						| by <<length-percentage>> ]
+		<<curve-command>> = curve
+						[ [ to <<position>> && with <<control-point>> [ / <<control-point>> ]? ]
+						| [ by <<coordinate-pair>> && with <<relative-control-point>> [ / <<relative-control-point>> ]? ] ]
+		<<smooth-command>> = smooth
+						[ [ to <<position>> && [with <<control-point>> ]? ]
+						| [ by <<coordinate-pair>> && [ with <<relative-control-point>> ]? ] ]
+		<<arc-command>> = arc <<command-end-point>>
+								&& of <<coordinate-pair>>
+								&& <<arc-sweep>>? && <<arc-size>>? && [rotate <<angle>>]?
+
+		<<command-end-point>> = [ to <<position>> | by <<coordinate-pair>> ]
+		<<control-point>> = [ <<position>> | <<relative-control-point>> ]
```

Comment:
When implementing this, I mentioned this to @smfr, and he didn't quite agree, but I thought I would bring it to a slightly larger audience.

While generally I don't really see the benefit in disallowing <position> to be used for `by` (since <position> is really just sugar for [ <length-percentage> <length-percentage> ], but if the distinction is going to be made, I think the grammar can be a bit more friendly:

```
<control-point> = [ <position> | <relative-control-point> ]
```

with one like so: 

```
<absolute-control-point> = <position> [ from [ start | end | origin ] ]?
```

or, if you dislike the duplication, of the `from` section,

```
<control-point-from> = [ from [ start | end | origin ] ]
<absolute-control-point> = <position> <control-point-from>?
<relative-control-point> = <coordinate-pair> <control-point-from>?
```

This allows a user who is already using an absolute control point, for example like so:

```
... curve to center center with top left ...
```

who decides that `with top left` (aka with 0% 0%) should really be `with top left from start` to do so. In the current grammar, they have to translate their top left to percentage form and write out  `with 0% 0% from start`.

(doing this also removes a slightly annoying, though manageable, ambiguity in the grammar when parsing `<<control-point>` since its not clear if "10% 10%" should match `<position>` or `<coordinate-pair>`.

Reply:
> When implementing this, I mentioned this to @smfr, and he didn't quite agree, but I thought I would bring it to a slightly larger audience.
> 
> While generally I don't really see the benefit in disallowing to be used for `by` (since is really just sugar for [ ], but if the distinction is going to be made, I think the grammar can be a bit more friendly:
> 
> ```
> <control-point> = [ <position> | <relative-control-point> ]
> ```
> 
> with one like so:
> 
> ```
> <absolute-control-point> = <position> [ from [ start | end | origin ] ]?
> ```
> 
> or, if you dislike the duplication, of the `from` section,
> 
> ```
> <control-point-from> = [ from [ start | end | origin ] ]
> <absolute-control-point> = <position> <control-point-from>?
> <relative-control-point> = <coordinate-pair> <control-point-from>?
> ```

This allows `to bottom left from end`, how would you interpret that?

Reply:
As identical to ‘to 100% 0% from end’?

Reply:
> As identical to ‘to 100% 0% from end’?

Ah so like transpose the reference rect to the start/end point... not sure it's a useful box for positioning, might create more confusion than usefulness.

Reply:
More confusing than the fact that ‘to 100% 0% from end’ is valid and ‘to bottom left’ is valid but not the merged form?

Mostly the point I am trying to make is that treating <position> as something other than syntax sugar for two <length- percentage>s seems undesirable. But changing this later is also non-breaking as far I can tell, so I really don’t want to push this any further than I have.

Reply:
> More confusing than the fact that ‘to 100% 0% from end’ is valid and ‘to bottom left’ is valid but not the merged form?

`to 100% 0% from end` is not valid, only `with 100% 0% from end` or `with 10px 220px / 100% 0% from end`, the `from start|end|origin` syntax is only for curve control points. 

I read `with bottom left` as a shorthand for `with 100% 0% from origin`, so expanding it to `with 100% 0% from origin from end` is indeed confusing.

> 
> Mostly the point I am trying to make is that treating as something other than syntax sugar for two s seems undesirable. But changing this later is also non-breaking as far I can tell, so I really don’t want to push this any further than I have.

Sounds good!

Reply:
Resolving this for now, feel free to file a new issue about this if you'd like.